geode-core/src/main/java/org/apache/geode/internal/cache/InitialImageOperation.java

```diff
@@ -1052,19 +1052,23 @@ protected RegionVersionVector getRVVFromProvider(final ClusterDistributionManage
   /**
    * Compare the received RVV with local RVV and return a set of keys for unfinished operations.
    *
-   * @param remoteRVV RVV from provider
+   * @param remoteRVV RVV from provider to be filled with unfinished operations
    * @param localRVV RVV recovered from disk
+   * @param receivedRVV original RVV from provider to remove departed members
    * @return set for keys of unfinished operations.
    */
   protected Set<Object> processReceivedRVV(RegionVersionVector remoteRVV,
-      RegionVersionVector localRVV) {
+      RegionVersionVector localRVV, RegionVersionVector receivedRVV) {
     if (remoteRVV == null) {
       return null;
     }
     // calculate keys for unfinished ops
+    Set<VersionSource> foundIds = new HashSet<>();
     HashSet<Object> keys = new HashSet<>();
-    if (region.getDataPolicy().withPersistence()
-        && localRVV.isNewerThanOrCanFillExceptionsFor(remoteRVV)) {
+    Set<VersionSource> departedMemberSet = receivedRVV.getDepartedMembersSet();
+    boolean isPersistentRegion = region.getDataPolicy().withPersistence();
+    if ((isPersistentRegion && localRVV.isNewerThanOrCanFillExceptionsFor(remoteRVV))
+        || !departedMemberSet.isEmpty()) {
```

Comment:
since foundIds is only added to if '!isPersistentRegion' it seems like this OR should be "|| (!isPersistentRegion && !departedMemberSet.isEmpty())" .
Otherwise we go into this "if" for a persistentRegion if departedMemberSet is not empty.

Reply:
departedMemberSet will always be empty for persistentRegion